src/classic/clvm_tools/stages/stage_2/optimize.rs

```diff
@@ -620,19 +664,34 @@ impl<'a> OptimizerRunner<'a> {
 
 pub fn optimize_sexp_<'a>(
     allocator: &mut Allocator,
+    memo: Rc<RefCell<HashMap<String, NodePtr>>>,
     r_: NodePtr,
     eval_f: Rc<dyn TRunProgram>,
 ) -> Result<NodePtr, EvalErr> {
     let mut r = r_;
```

Comment:
unrelated to this patch, but this variable could have a smaller scope by being moved down to the top of the loop where it's used. I think it would be especially helpful since it's mutable, it adds a lot of cognitive load to the code below